fundamentals/android/auth-sign-in-with-google/Notes/app/src/main/java/com/notes/app/screens/authentication/sign_in/SignInScreen.kt

```diff
@@ -159,7 +160,7 @@ fun SignInScreen(
             .fillMaxWidth()
             .padding(8.dp))
 
-        TextButton(onClick = { viewModel.onSignUpClick(onSignUpClicked) }) {
+        TextButton(onClick = { onSignUpClicked(SIGN_UP_SCREEN) }) {
```

Comment:
Do we need this argument `SIGN_UP_SCREEN`? What other values can be passed to `onSignUpClicked` and what do they represent?

Reply:
Yes we need to pass on the name of the screen we're navigating to. This "onSignUpClicked" is just the name I chose for this parameter (to match the context in this screen), but in truth it is the `navigate` method available in `NotesAppState` (that takes a route as parameter).